### ISSUE TITLE
Crash when resizing window with multiple terminals open

### ISSUE THREAD

**Issue by thierer** [gnunn1/tilix#1028]
Tilix crashes for me nearly 100% when resizing a window with multiple (>1) terminals open, taking all open (tilix) windows with it. Resizing a window with only one terminal works fine.

This is an up-to-date Arch Linux install running xfce on x64_64. Tilix and vte3 are installed from the [tilix-bin ](https://aur.archlinux.org/packages/tilix-bin/) and [vte3-tilix](https://aur.archlinux.org/packages/vte3-tilix/) packages on AUR.

Is there anything I can do to debug this? I tried starting tilix from a (non-tilix) terminal, but there is no message when it crashes.

**Comment by gnunn1** [comment]
Easiest way to debug it is too build tilix from source and then use gdb to get a back trace. I'm out of the house today but when I get back tonite I can build you a debug version to test with.

**Comment by thierer** [comment]
I found the build instructions a bit scary so I was reluctant to build it myself, but I tried it and it wasn't too bad :)

Here's what I did:

1. Cloned tilix and checked out version 1.6.4 (because that's the version I have installed)
2. Installed dmd, libphobos and dtools from the Arch repository
3. Installed dub from [AUR](https://aur.archlinux.org/packages/dub/)
4. Built tilix with `dub build --build=debug`
5. **I did not install the files from the build**, because I didn't wan't to overwrite those from my original install and I figured they should be identical as both should be v 1.6.4
6. `gdb ./tilix` and `run` from within the build directory
7. In tilix: Split the Terminal with `Ctrl-Alt-D` and vertically resized the window by dragging the bottom border with the mouse until it crashed
8. `bt` inside gdb.

 I'll attach the output of the gdb including the backtrace. I tried three times and the ouput looked identical at first glance.

[gdb output.txt](https://github.com/gnunn1/tilix/files/1167340/gdb.output.txt)

Let me know if you need anything else. Just be aware that have no experience with D and next to no experience with glib and gtk... :)

**Comment by gnunn1** [comment]
Thanks for taking the time to do this, unfortunately that backtrace doesn't help me since the backtrace is all in GTK stuff rather then tilix. One thing you can try, in session.d try commenting out the addOnSizeAllocate lines at 1551 and see if it makes any difference.

So change this:
```
        addOnSizeAllocate(delegate(GdkRectangle* rect, Widget) {
            updatePosition();
        });
```

to this:

```
/*
        addOnSizeAllocate(delegate(GdkRectangle* rect, Widget) {
            updatePosition();
        });
*/
```

**Comment by gnunn1** [comment]
Also note that I do not officially support the xfce desktop, I'll try to help you as best I can but I'm not willing to try to reproduce it unless it is occurring in Gnome or Unity.

**Comment by thierer** [comment]
I tried commenting out the `addOnSizeAllocate()` call like you suggested, but unfortunately it still crashes.

Thanks for your help, anyway! I guess I'd stick to my current workaround of resizing the window before opening additional terminals, unless you have any suggestions what else I could try.

Anyway, a few more observations:

1. I don't **think** the problem existed (but maybe I was just lucky) when I started using tilix (v1.5.4), so I tried downgrading to that version, but it also crashes which suggests the problem is not in tilix. I also downgraded glib and gtk to its contemporary versions, but that didn't change anything, neither.

2. I noticed that it does not crash when I resize the window **really slow** (sorry, I couldn't really come up with an objective description) which suggests it's a race condition somewhere.
